### DISCUSSION TITLE
Can't get "clientAuthentication is not a function" to go away

### DISCUSSION THREAD

**Discussion by jordan-dalby** [panva/openid-client#726] [Q&A] [answered]
Hi,
Sorry if it's been asked a million times, I can't figure this out. Trying to integrate SSO with OIDC (using Authentik for testing) and running into the issue in the title.
```
server-1  | [ERROR] OIDC callback error: TypeError: clientAuthentication is not a function
server-1  |     at authenticatedRequest (file:///app/node_modules/oauth4webapi/build/index.js:1050:11)                              
server-1  |     at tokenEndpointRequest (file:///app/node_modules/oauth4webapi/build/index.js:1069:28)                              
server-1  |     at Module.authorizationCodeGrantRequest (file:///app/node_modules/oauth4webapi/build/index.js:1268:12)              
server-1  |     at Module.authorizationCodeGrant (file:///app/node_modules/openid-client/build/index.js:649:10)
server-1  |     at OIDCConfig.handleCallback (file:///app/src/oidc/oidcConfig.js:150:33)                                            
server-1  |     at file:///app/src/routes/oidcRoutes.js:50:45                                                                       
server-1  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) 
```
Here are the files in question, the issue is with `fetchUserInfo` supposedly, can't see where though, is it somewhere internal?
https://gist.github.com/jordan-dalby/bcdd438d2f3b99a347533fe569a20d0b
https://gist.github.com/jordan-dalby/f0cf61eb11f4d8992f7aeb9a1a725b74

Any help appreciated, cheers.

**Comment by panva** [comment, marked as answer]
From a cursory look you just need to completely remove the `options` object that isn't a valid value for the `clientAuthentication` argument of [`client.discovery()`](https://github.com/panva/openid-client/blob/v6.1.3/docs/functions/discovery.md#parameters)

**Reply by jordan-dalby** [reply]
Doh. Thank you.
